modules/exploits/multi/http/gitlab_file_read_rce.rb

```diff
@@ -242,8 +242,12 @@ def create_issue(project:, issue:)
       elsif res.code != 302
         raise GitLabClientException, "Unexpected HTTP #{res.code} response."
       end
-
-      issue_id = res.body[%r{You are being <a href="http://.*#{create_issue_path}/(\d+)">redirected</a>}, 1]
+      
+      if http_client.ssl
+        issue_id = res.body[%r{You are being <a href="https://.*#{create_issue_path}/(\d+)">redirected</a>}, 1]
+      else
+        issue_id = res.body[%r{You are being <a href="http://.*#{create_issue_path}/(\d+)">redirected</a>}, 1]
+      end
```

Comment:
What are your thoughts on updating the regex to allow an optional `s` ? 👀 

I haven't tested it yet, but it seems sane 😄 

```suggestion
issue_id = res.body[%r{You are being <a href="https?://.*#{create_issue_path}/(\d+)">redirected</a>}, 1]
```